src/ui/pages/CreateSSIAgent/CreateSSIAgent.tsx

```diff
@@ -0,0 +1,226 @@
+import { IonButton, IonIcon, IonSpinner } from "@ionic/react";
+import { informationCircleOutline, scanOutline } from "ionicons/icons";
+import { useState, MouseEvent as ReactMouseEvent, useMemo } from "react";
+import { i18n } from "../../../i18n";
+import { RoutePath } from "../../../routes";
+import { getNextRoute } from "../../../routes/nextRoute";
+import { useAppDispatch, useAppSelector } from "../../../store/hooks";
+import {
+  getStateCache,
+  setCurrentOperation,
+} from "../../../store/reducers/stateCache";
+import { updateReduxState } from "../../../store/utils";
+import { CustomInput } from "../../components/CustomInput";
+import { ErrorMessage } from "../../components/ErrorMessage";
+import { PageFooter } from "../../components/PageFooter";
+import { PageHeader } from "../../components/PageHeader";
+import { OperationType } from "../../globals/types";
+import { useAppIonRouter } from "../../hooks";
+import "./CreateSSIAgent.scss";
+import {
+  clearSSIAgent,
+  getSSIAgent,
+  setBootUrl,
+  setConnectUrl,
+} from "../../../store/reducers/ssiAgent";
+import { isValidHttpUrl } from "../../utils/urlChecker";
+import { TermsModal } from "../../components/TermsModal";
+import { Agent } from "../../../core/agent/agent";
+import { ScrollablePageLayout } from "../../components/layout/ScrollablePageLayout";
+
+const CreateSSIAgent = () => {
+  const pageId = "create-ssi-agent";
+  const ssiAgent = useAppSelector(getSSIAgent);
+
+  const stateCache = useAppSelector(getStateCache);
+  const ionRouter = useAppIonRouter();
+  const dispatch = useAppDispatch();
+  const [connectUrlInputTouched, setConnectUrlTouched] = useState(false);
+  const [bootUrlInputTouched, setBootUrlInputTouched] = useState(false);
+  const [openInfo, setOpenInfo] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [hasMismatchError, setHasMismatchError] = useState(false);
+
+  const setTouchedConnectUrlInput = () => {
+    setConnectUrlTouched(true);
+  };
+
+  const setTouchedBootUrlInput = () => {
+    setBootUrlInputTouched(true);
+  };
+
+  const validBootUrl = useMemo(() => {
+    return ssiAgent.bootUrl && isValidHttpUrl(ssiAgent.bootUrl);
+  }, [ssiAgent]);
+
+  const validConnectUrl = useMemo(() => {
+    return ssiAgent.connectUrl && isValidHttpUrl(ssiAgent.connectUrl);
+  }, [ssiAgent]);
+
+  const displayBootUrlError =
+    bootUrlInputTouched &&
+    ssiAgent.bootUrl &&
+    !isValidHttpUrl(ssiAgent.bootUrl);
+
+  const displayConnectUrlError =
+    connectUrlInputTouched &&
+    ssiAgent.connectUrl &&
+    !isValidHttpUrl(ssiAgent.connectUrl);
+
+  const validated = validBootUrl && validConnectUrl && !hasMismatchError;
+
+  const handleClearState = () => {
+    dispatch(clearSSIAgent());
+  };
+
+  const handleValidate = async () => {
+    setLoading(true);
+    try {
+      await Agent.agent.bootAndConnect({
+        bootUrl: ssiAgent.bootUrl || "",
```

Comment:
I think it makes more sense to do a check first to make sure `ssiAgent.bootUrl` and `ssiAgent.connectUrl` aren't undefined and throw an error or something.

Otherwise this might get hard to debug when e.g. the connectUrl isn't being updated correctly in Redux